tor/controller.go

```diff
@@ -350,46 +380,38 @@ func supportsV3(version string) error {
 
 // ProtocolInfo returns the different authentication methods supported by the
 // Tor server and the version of the Tor server.
-func (c *Controller) ProtocolInfo() ([]string, string, string, error) {
+func (c *Controller) ProtocolInfo() ([]string, string, map[string]string, error) {
```

Comment:
Would prefer if this just returned all of the authentication methods supported rather than the whole info response.

Reply:
The supported auth methods is the first return value (the `[]string`). But depending on the method, the actual auth implementation may require additional stuff from the PROTOCOLINFO response. For example the SAFECOOKIE method requires the COOKIEFILE part to know where to load the cookie from (search in the code for `info["COOKIEFILE"]`).

Reply:
Gotcha, in that case I think we should bundle all of these return arguments into a struct for easier readability.